service/module.go

```diff
@@ -47,58 +45,58 @@ type Module interface {
 }
 
 // ModuleProviderFromFunc creates a new ModuleProvider from a name and create function.
-func ModuleProviderFromFunc(name string, createFunc func(Host) (Module, error)) ModuleProvider {
-	return &moduleProviderFunc{name, createFunc}
+func ModuleProviderFromFunc(createFunc func(Host) (Module, error)) ModuleProvider {
+	return &moduleProvider{createFunc: createFunc}
 }
 
-// ModuleOption is a function that configures module creation.
-type ModuleOption func(*moduleOptions) error
+// ModuleOptionFn is a function that configures module creation.
+type ModuleOptionFn func(*moduleOption) error
 
-// WithModuleName will override the name given by the ModuleProvider.
-func WithModuleName(name string) ModuleOption {
-	return func(o *moduleOptions) error {
-		o.name = name
+// WithName will override the name given by the ModuleProvider.
+func WithName(name string) ModuleOptionFn {
```

Comment:
I think this should still be WithModuleName - you have WithModule (not With) on service, and there is more than one construct in the service package.